resource/ui/ts/pages/nodes.ts

```diff
@@ -13,14 +15,66 @@ module AdminViews {
    * Nodes is the view for exploring the status of all nodes.
    */
   export module Nodes {
+    // TODO(bram): Lots of duplicate code between here and stores.ts, refactor
+    // it into one place.
+    export interface RefreshFunction {
+      f: () => _mithril.MithrilPromise<any>;
+      o: any;
+    }
+
+    export interface QueryManagerSourceMap {
+      [nodeId: number]: { [source: string]: Models.Metrics.QueryManager }
+    }
+
     export var nodeStatuses: Models.NodeStatus.Nodes = new Models.NodeStatus.Nodes();
+    export var queryManagers: QueryManagerSourceMap = {};
+
     export class Controller implements _mithril.MithrilController {
       private static _queryEveryMS = 10000;
       private _interval: number;
+      private _nodeId: string;
+      private _refreshFunctions: RefreshFunction[]
+
+      private _refresh():void {
+        for (var i = 0; i < this._refreshFunctions.length; i++) {
+          this._refreshFunctions[i].f.call(this._refreshFunctions[i].o);
+        }
+      }
+
+      private static _queryManagerBuilder(nodeId: string, agg: Models.Metrics.QueryAggregator, source:string): Models.Metrics.QueryManager {
```

Comment:
This is the fancy JS I had to come up with to get multiple charts working in a sane manner.  
The builder creates instances of the queryManager (each one essentially represents a single chart on a page at this point.)
The _refresh function is where all the requery logic sits, to update the data every interval.